hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/SplitBrainJoinMessage.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.cluster.impl;
+
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.version.Version;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * A {@code JoinMessage} issued by the master node of a subcluster to the master of another subcluster
+ * while searching for other clusters for split brain recovery.
+ */
+public class SplitBrainJoinMessage extends JoinMessage {
+
+    protected Version clusterVersion;
+
```

Comment:
It is sent in `JoinMessage.version`, however when attempting to locate clusters for merge, the master node's codebase version is irrelevant in this implementation.